pkg/controller/atlasproject/backupcompliancepolicy.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2023 MongoDB.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package atlasproject
+
+import (
+	"errors"
+	"net/http"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/cmp"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api"
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/controller/workflow"
+)
+
+const bcpNotMet = "BACKUP_POLICIES_NOT_MEETING_BACKUP_COMPLIANCE_POLICY_REQUIREMENTS"
+
+type backupComplianceController struct {
+	ctx     *workflow.Context
+	client  client.Client
+	project *akov2.AtlasProject
+}
+
+func (r *AtlasProjectReconciler) ensureBackupCompliance(ctx *workflow.Context, project *akov2.AtlasProject) workflow.Result {
+	ctx.Log.Debug("starting backup compliance policy processing")
+	defer ctx.Log.Debug("finished backup compliance policy processing")
+
+	b := backupComplianceController{
+		ctx:     ctx,
+		client:  r.Client,
+		project: project,
+	}
+
+	return b.handlePending()
+}
+
+func (b *backupComplianceController) handlePending() workflow.Result {
+	bcp, found, err := b.getAtlasBackupCompliancePolicy()
+	if err != nil {
+		return b.terminate(workflow.Internal, err)
+	}
+
+	akoEmpty := true
+	if bcpRef := b.project.Spec.BackupCompliancePolicyRef; bcpRef != nil {
+		akoEmpty = bcpRef.IsEmpty()
+	}
+
+	atlasEmpty := !found
+
+	switch {
+	case !akoEmpty && atlasEmpty:
+		return b.upsert(bcp)
+	case !akoEmpty && !atlasEmpty:
+		return b.upsert(bcp)
+	case akoEmpty && !atlasEmpty:
+		return b.delete()
+	default:
+		return b.unmanage()
```

Comment:
nitpick, I believe inverting the env vars name and semantic makes the cases more readable:

```suggestion
	case inAKO && !inAtlas:
		return b.upsert(bcp)
	case inAKO && inAtlas:
		return b.upsert(bcp)
	case !inAKO && inAtlas:
		return b.delete()
	default:
		return b.unmanage()
```

Reply:
I think you're right, nicer to parse